### PR TITLE
Update tasks to azure-pipelines-task-lib 2.8.0

### PR DESCRIPTION
**Pull request by joshmgross**
Updating a bunch of related tasks to `azure-pipelines-task-lib 2.8.0`

Change list:
- ANTV1
  - Updated to `azure-pipelines-task-lib 2.8.0`
- GradleV2
  - Updated to `azure-pipelines-task-lib 2.8.0`
  - Removed dependency on `Common/Utility-Common` (`getSystemAccessToken`
- MavenV2
  - Updated to `azure-pipelines-task-lib 2.8.0`
  - Switched from `pacakaging-common-old` to `packaging-common`
- MavenV3
  - Updated to `azure-pipelines-task-lib 2.8.0`
  - Switched from `pacakaging-common-old` to `packaging-common`
  - Ported over legacy tests
- XamarinAndroidV1
  - Updated to `azure-pipelines-task-lib 2.8.0`
  - Removed dependency on `Common/MSBuildHelpers` (`getMSBuildPath`)
  - Ported over legacy tests
- PublishCodeCoverageResultsV1
  - Bumped version for changes in `Common\codecoverage-tools`
  - A resource string was automatically updated as well

- Common tasks updated to `azure-pipelines-task-lib 2.8.0`:
  - codecoverage-tools
  - codeanalysis-common
  - java-common

- packaging-common-old
  - Removed as it was only used by MavenV2 and MavenV3 while they were on the old task lib version

Tested via build canaries for ANTV1, GradleV2, MavenV2, MavenV3, XamarinAndroidV1, and PublishCodeCoverageResultsV1

### CODE REVIEW

**Comment by joshmgross** [comment]
I ported over the legacy tests for MavenV3 and XamarinAndroid. Most of them are intact, a few I deemed were duplicate and not ported over. For Maven, I left out a few that weren't really testing anything MavenV3 task-related (such as common package tests) as those are now mocked out.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -119,8 +119,6 @@ Tasks/Common/java-common/*        @madhurig   @kelliejos
 
 Tasks/Common/kubernetes-common/*      @nishubansal    @ds-ms
 
-Tasks/Common/packaging-common-old/*   @jotaylo
```
Comment (jotaylo) [inline .github/CODEOWNERS]: Signing off on the removal of this folder (thank you!)

File: Tests-Legacy/L0/CodeAnalysis-Common/_suite.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
+/// <reference path="../../definitions/mocha.d.ts"/>
```
Comment (brcrista) [inline Tests-Legacy/L0/CodeAnalysis-Common/_suite.ts]: how come we're adding a legacy test?

Comment (joshmgross) [reply Tests-Legacy/L0/CodeAnalysis-Common/_suite.ts]: These tests were in the Maven legacy tests and don't need to be ported over.

**Pull request merged**